projects/plugins/protect/src/js/data/scan/use-fixers-mutation.ts

```diff
@@ -15,9 +16,32 @@ export default function useFixersMutation(): UseMutationResult {
 
 	return useMutation( {
 		mutationFn: API.fixThreats,
-		onSuccess: data => {
-			// The data returned from the API is the same as the data we need to update the cache.
-			queryClient.setQueryData( [ QUERY_FIXERS_KEY ], data );
```

Comment:
Is the data returned from the `fixThreats` endpoint not usable here?

If we do need to optimistically update the cache:
- We could optimistically update right away, i.e. in `onMutate` instead of `onSuccess`.
- We should also add a graceful revert in the mutation's `onError` callback (examples using react query's "context" variable should be in other mutation hooks).

Reply:
>Is the data returned from the fixThreats endpoint not usable here?
It wasn't for me, but I am realizing just now as you mention this that the way I was modifying the API method for fixing threats so I could test on Jurassic Tube was quite possibly the cause of that. I'll test it out on a JN instance with the original approach and see if it works! Would be a nice simplification.